docpipeline/generate.py

```diff
@@ -89,7 +89,11 @@ def process_blob(blob, credentials, devsite_template):
     tar.decompress(tar_filename, api_path)
     log.info(f"Decompressed {blob.name} in {api_path}")
 
-    metadata_path = api_path.joinpath("docs.metadata")
+    metadata_file = "docs.metadata"
+    if api_path.joinpath("docs.metadata.json").exists():
+        metadata_file = "docs.metadata.json"
+
+    metadata_path = api_path.joinpath(metadata_file)
     metadata = metadata_pb2.Metadata()
     text_format.Merge(metadata_path.read_text(), metadata)
```

Comment:
This seems to parse with `text_format` either way. `text_format` expects the content to be in text proto format (https://googleapis.dev/python/protobuf/latest/google/protobuf/text_format.html). I think we need to use `json_format` when the file is JSON.

Something like:
```python
from google.protobuf import text_format, json_format

# ...

metadata = metadata_pb2.Metadata()
json_file = api_path.joinpath("docs.metadata.json")
if json_file.exists():
    # Use json_format to parse json_file into metadata.
else:
    # Use text_format.
```

Reply:
I added it in docuploader but missed it out on here. :(
Thank you for catching it!